src/main/java/hellfirepvp/astralsorcery/common/item/wand/ItemExchangeWand.java

```diff
@@ -211,22 +211,17 @@ public ActionResultType onItemUse(ItemUseContext context) {
             }
 
             BlockState prevState = world.getBlockState(placePos);
-            if (((ServerPlayerEntity) player).interactionManager.tryHarvestBlock(placePos)) {
-                if (MiscUtils.canPlayerPlaceBlockPos(player, stateToPlace, placePos, Direction.UP)) {
-                    if (AlignmentChargeHandler.INSTANCE.drainCharge(player, LogicalSide.SERVER, COST_PER_EXCHANGE, false) &&
-                            world.setBlockState(placePos, stateToPlace)) {
-                        if (!player.isCreative()) {
-                            ItemUtils.consumeFromPlayerInventory(player, stack, extractable, false);
-                        }
-
-                        PktPlayEffect ev = new PktPlayEffect(PktPlayEffect.Type.BLOCK_EFFECT)
-                                .addData(buf -> {
-                                    ByteBufUtils.writePos(buf, placePos);
-                                    ByteBufUtils.writeBlockState(buf, prevState);
-                                });
-                        PacketChannel.CHANNEL.sendToAllAround(ev, PacketChannel.pointFromPos(world, placePos, 32));
-                    }
-                }
+            if ((player.isCreative() || ItemUtils.consumeFromPlayerInventory(player, stack, extractable, false)) &&
```

Comment:
I'd like to note that this is going on a bit of a stretch here.
Assuming that you do have a item, but not enough starlight change - for example after (re)placing ~100 blocks, this would consume one additional one, before checking at all if there's enough charge left.

So i'll rewrite this myself to do the most checks possible to ensure it is gonna be valid before actually performing item or other consumption.

Reply:
Right on - if only minecraft had transactions this would be so much easier ;)